openedx/core/djangoapps/verified_track_content/partition_scheme.py

```diff
@@ -0,0 +1,147 @@
+        """
+        course_key = CourseKey.from_string(self.parameters["course_id"])
+
+        if is_course_using_cohort_instead(course_key):
+            return []
+
+        all_groups = []
+        for mode in CourseMode.modes_for_course(course_key, include_expired=True, only_selectable=False):
+            group = Group(ENROLLMENT_GROUP_IDS[mode.slug], unicode(mode.name))
+            all_groups.append(group)
+
+        return all_groups
+
+    def to_json(self):
+        """
+        'Serialize' to a json-serializable representation.
+
+        Returns:
+            a dictionary with keys for the properties of the partition, with an empty array for
+            `groups` because it is dynamic based on the current course modes.
+        """
+        return {
+            "id": self.id,
+            "name": self.name,
+            "scheme": self.scheme.name,
+            "description": self.description,
+            "parameters": self.parameters,
+            "groups": [],  # Do not persist groups because they are dynamic.
+            "active": bool(self.active),
+            "version": UserPartition.VERSION
+        }
+
+
+class EnrollmentTrackPartitionScheme(object):
+    """
+    This scheme uses learner enrollment tracks to map learners into partition groups.
+    """
+
+    @classmethod
+    def get_group_for_user(cls, course_key, user, user_partition):
+        """
+        Returns the Group from the specified user partition to which the user
+        is assigned, via enrollment mode.
+        """
+        # NOTE: masquerade code was copied from CohortPartitionScheme, and it may need
+        # some changes (or if not, code should be refactored out and shared).
+        # This work will be done in a future story TNL-6739.
+
+        # First, check if we have to deal with masquerading.
+        # If the current user is masquerading as a specific student, use the
+        # same logic as normal to return that student's group. If the current
+        # user is masquerading as a generic student in a specific group, then
+        # return that group.
+        if get_course_masquerade(user, course_key) and not is_masquerading_as_specific_student(user, course_key):
+            group_id, user_partition_id = get_masquerading_group_info(user, course_key)
+            if user_partition_id == user_partition.id and group_id is not None:
+                try:
+                    return user_partition.get_group(group_id)
+                except NoSuchUserPartitionGroupError:
+                    return None
+            # The user is masquerading as a generic student. We can't show any particular group.
+            return None
+
+        if is_course_using_cohort_instead(course_key):
+            return None
+        mode_slug, is_active = CourseEnrollment.enrollment_mode_for_user(user, course_key)
+        if mode_slug and is_active:
+            course_mode = CourseMode.mode_for_course(course_key, mode_slug)
```

Comment:
This may not work for old courses (as [tagged on the bulk-email PR](https://github.com/edx/edx-platform/pull/14688/files#r105997547)).  The implementation of `mode_for_course` calls `modes_for_course` with the limiting default values for `include_expired` and `only_selectable`.